### ISSUE TITLE
Should we allow unrecognized fields for the signatures dictionaries?

### ISSUE THREAD

**Issue by MVrachev** [theupdateframework/tuf#1802]
**Description of issue or feature request**:
In [ADR 8](https://github.com/theupdateframework/python-tuf/blob/develop/docs/adr/0008-accept-unrecognised-fields.md) we follow the [Document formats](https://theupdateframework.github.io/specification/latest/#document-formats) section of the specification and allow unrecognized fields at each level of the metadata files.
The question is do we want to allow unrecognized fields inside the `signatures` dictionaries? 
Are there possible security issues to that?

For context see the specification issue on the matter: https://github.com/theupdateframework/specification/issues/203

**Comment by jku** [comment]
Lets decide this before 1.0 release (with the understanding that we might just decide to postpone)

**Comment by jku** [comment]
Quoting myself in sslib issue: 

> It is true that we should be more cautious with this part of the metadata (not because they contain signatures but because they are not part of the signed portion of metadata). But: because of TUF design that requires canonicalization we end up parsing even the signed portion before we can check that it has been signed.

Because of this _I think_ allowing unrecognised fields in signature does not significantly increase risk


Oh and the reason why we might want to do this: I believe supporting fulcio certificates/signatures requires extending the metadata. If unrecognised fields are not supported that would mean python-tuf 1.0 might not be able to read metadata written by python-tuf 1.x with fulcio support

**Comment by mnm678** [comment]
cc @asraa who may have insight into the Fulcio signatures.

When using canonical json, it's true that adding an additional field doesn't increase the risk, because we are already doing some somewhat risky parsing. If we switch to a format like DSSE, we'll want to avoid adding additional fields to the signature that could make the implementation vulnerable to a [canonicalization attack](https://soatok.blog/2021/07/30/canonicalization-attacks-against-macs-and-signatures/).

**Comment by jku** [comment]
To complete the options here: If we allow unrecognised_fields in Signature then there's nothing preventing allowing them in Metadata itself as well.

**Comment by asraa** [comment]
> cc @asraa who may have insight into the Fulcio signatures.

It was majorly convenient adding a cert field to the signature -- +1

**Comment by MVrachev** [comment]
Together in a discussion with @jku and @lukpueh we agreed that it makes sense to implement unrecognized fields inside `Metadata`, `Metadata.signatures` and in the `Signatures` objects.
You can see read more about my reasoning here: https://github.com/theupdateframework/specification/issues/203#issuecomment-1031479746.

**Comment by jku** [comment]
`Metadata.signatures`: this does not make sense. signatures is an array and the items are clearly defined as Signature objects (Signature objects are not named in the specification but the content is defined).

**Comment by MVrachev** [comment]
Okay, my bad, I misunderstood.
Thanks for the clarification.

**Comment by MVrachev** [comment]
I will close this issue as it's related to a `securesystemslib` code and I will create a new one for `Metadata` unrecognized fields.
